src/riak_cs_gc_manager.erl

```diff
@@ -0,0 +1,325 @@
+
+%% @doc Adjust the interval at which the daemon attempts to perform
+%% a garbage collection sweep. Setting the interval to a value of
+%% `infinity' effectively disable garbage collection. The daemon still
+%% runs, but does not carry out any file deletion.
+-spec set_interval(term()) -> ok | {error, term()}.
+set_interval(Interval) when is_integer(Interval)
+                            orelse Interval =:= infinity ->
+    gen_fsm:sync_send_all_state_event(?SERVER, {set_interval, Interval}, infinity).
+
+start_link() ->
+    gen_fsm:start_link({local, ?SERVER}, ?MODULE, [], []).
+
+-ifdef(TEST).
+test_link() ->
+    gen_fsm:start_link({local, ?SERVER}, ?MODULE, [testing], []).
+
+-endif.
+
+
+finished(Report) ->
+    gen_fsm:sync_send_event(?SERVER, {finished, Report}, infinity).
+
+%%%===================================================================
+%%% gen_fsm callbacks
+%%%===================================================================
+
+init([]) ->
+    process_flag(trap_exit, true),
+    InitialDelay = riak_cs_gc:initial_gc_delay(),
+    State = case riak_cs_gc:gc_interval() of
+                infinity ->
+                    #gc_manager_state{};
+                Interval when is_integer(Interval) ->
+                    Interval2 = Interval + InitialDelay,
+                    Next = riak_cs_gc:timestamp() + Interval2,
+                    TimerRef = erlang:send_after(Interval2 * 1000, self(), {start, []}),
+                    _ = lager:info("Scheduled next batch at ~s",
+                                   [riak_cs_gc_console:human_time(Next)]),
+
+                    #gc_manager_state{next=Next,
+                                      interval=Interval,
+                                      initial_delay=InitialDelay,
+                                      timer_ref=TimerRef}
+        end,
+    {ok, idle, State};
+init([testing]) ->
+    {ok, idle, #gc_manager_state{}}.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc All gen_fsm:send_event/2 call should be ignored
+idle(_Event, State) ->
+    {next_state, idle, State}.
+running(_Event, State) ->
+    {next_state, running, State}.
+
+%% @private
+%% @doc
+idle({start, Options}, _From, State) ->
+    case start_batch(State, Options) of
+        {ok, NextState} ->
+            {reply, ok, running, NextState};
+        Error ->
+            {reply, Error, idle, State}
+    end;
+idle(_, _From, State) ->
+    {reply, {error, idle}, idle, State}.
+
+running(cancel, _From, State = #gc_manager_state{gc_batch_pid=Pid}) ->
+    %% stop gc_batch here
+    catch riak_cs_gc_batch:stop(Pid),
+    NextState = schedule_next(State),
+    {reply, ok, idle, NextState#gc_manager_state{gc_batch_pid=undefined}};
+running({finished, Report}, _From, State) ->
+    %% Add report to history
+    NextState=schedule_next(State),
+    {reply, ok, idle,
+     NextState#gc_manager_state{batch_history=[Report]}};
```

Comment:
No. `gc_batch_pid` is set undefined at `handle_info` as the process traps exits.